lightning/src/onion_message/messenger.rs

```diff
@@ -744,6 +789,27 @@ where
 		)
 	}
 
+	/// Forwards an [`OnionMessage`] to `peer_node_id`. Useful if we initialized
+	/// the [`OnionMessenger`] with [`Self::new_with_offline_peer_interception`]
+	/// and want to forward a previously intercepted onion message to a peer that
+	/// has just come online.
+	pub fn forward_onion_message(
+		&self, message: OnionMessage, peer_node_id: &PublicKey
+	) -> Result<(), SendError> {
+		let mut message_recipients = self.message_recipients.lock().unwrap();
+		if outbound_buffer_full(&peer_node_id, &message_recipients) {
+			return Err(SendError::BufferFull);
+		}
+
+		match message_recipients.entry(*peer_node_id) {
+			hash_map::Entry::Occupied(mut e) if e.get().is_connected() => {
+				e.get_mut().enqueue_message(message);
+				Ok(())
+			},
+			_ => Err(SendError::InvalidFirstHop(*peer_node_id))
```

Comment:
I was giving a thought on how we are handling this situation. It might be possible that the peer is known, but just not connected at the instant. Sending an `InvalidFirstHop` error might confuse users since it reads “**The first hop is not a peer and …**”. Maybe we could introduce a new Error type for cases like this?

Reply:
Hmm, why is that confusing? At this stage we don't have any way of knowing whether the peer is known or not. I think "not a peer" captures that pretty well?

Reply:
If we're not sure if the peer's known or not, `InvalidFirstHop` could work. But in this code snippet, we're checking if the peer's connected with `if e.get().is_connected()`, not if it's known. So, something like a `PeerNotConnected` error would make more sense for the error message. Still, I'm cool with whatever you think works best!